packages/auth/src/session.ts

```diff
@@ -0,0 +1,34 @@
+import { MINTBASE_CONNECT_HOST } from './constants';
+import { MintbaseSession } from './wallet';
+
+
+type RequestLike = {
+  body: object;
+}
+
+export const proxySessionTokenRequest = async (req: RequestLike, useEndpoint?: string): Promise<string> => {
+  const session = await fetch(useEndpoint || `${MINTBASE_CONNECT_HOST}/auth`, {
+    method: 'POST',
+    headers: {
+      'mb-api-key': 'anon',
+      'content-type':'application/json',
+    },
+    body: JSON.stringify(req.body),
+  });
+  const { token } = await session.json();
+  return token;
+};
+
+export const validateSessionToken = async (token: string, useEndpoint?: string): Promise<MintbaseSession> => {
```

Comment:
This method in particular can be used by any endpoint, anywhere in the services layer including vercel endpoints.